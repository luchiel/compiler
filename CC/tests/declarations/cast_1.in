void main()
{
int* i;
float j[4];
i = (int*)j;
}