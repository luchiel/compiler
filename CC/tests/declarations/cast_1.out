type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function printf
	returns type <int>
	arguments
		<no args>
function main
	returns type <void>
	arguments
		<no args>
	{{}}
		type <0>
			pointer to <int>
		variable i of type <0>
		type <1>
			array [
				{int 4}
					type <int>
			] of <float>
		variable j of type <1>
		type <2>
			pointer to <int>
	| 
	+-{;}
	  | 
	  +-{=}
				type <0>
					pointer to <int>
	    | 
	    +-{i}
					type <0>
						pointer to <int>
	    | 
	    +-{cast}
					type <2>
						pointer to <int>
	      | 
	      +-<2>
	      | 
	      +-{j}
						type <1>
							array [
								{int 4}
									type <int>
							] of <float>
